src/app/static/src/components/parameters/Parameters.vue

```diff
@@ -102,13 +130,30 @@ export default defineComponent({
             context.emit("updateValues", newParamValues);
         }
 
+        function updatePhases(newPhases: Rt[]) {
+            const groupId = editParamGroupId.value;
+            closePhases();
+            const idx = props.paramGroupMetadata.findIndex((g) => g.id === groupId);
+            const newMetadata = [...props.paramGroupMetadata];
+            newMetadata[idx] = { ...newMetadata[idx], config: newPhases };
+            context.emit("updateMetadata", newMetadata);
```

Comment:
If I'm understanding correctly then `updateMetadata` is required to update the charts, and `updateValues` to update the sidebar and retrieve new data?

Reply:
Almost - the metadata is what's provided to the Parameter views in the sidebar, both the standard dynamic form views and the Phases views. In the case of the form, the metadata is of a different format to values, since it includes all the form display stuff, rather than just being a dictionary of values, but they're the same thing for Phases (so we wouldn't need the distinction if not for this oddity of the dynamic form). So updating the metadata feeds the new values back into the sidebar. Updating values fetches new results, causing the charts to update.